presto-main/src/test/java/io/prestosql/operator/scalar/timestamptz/TestExtract.java

```diff
@@ -181,6 +183,25 @@ public void testDayOfWeek()
         assertThat(assertions.expression("day_of_week(TIMESTAMP '2020-05-10 12:34:56.123456789012 Asia/Kathmandu')")).matches("BIGINT '7'");
     }
 
+    @Test
+    public void testDayOfWeekWithWeekStart()
+    {
```

Comment:
I think we should also test the cases with higher precision (> 3), like the `testDayOfWeek` method. 

For example, the issue in `ExtractDayOfWeekWithWeekStart` for timstamptz implementation would've been caught with such a testcase.

We don't need to add 8 * 13 tests, but one for a different codepath should be sufficient, I think.

Reply:
same comment for the test in `io.prestosql.operator.scalar.timestamp.TestExtract` class.

Reply:
True, we could have caught the missing part in timestamptz. Added test cases for higher precision and covered all the cases of our logic, let me know if I am missing something here.